ApiClient/Source/Pages/Queries/SearchQuery.swift

```diff
@@ -27,22 +27,26 @@ import Foundation
 import Alamofire
 
 final class SearchQuery: Query {
-    typealias Result = JSONPage?
+    typealias Result = JSONPage
     
-    let resource: String
+    let url: URL?
+    let host: String = "https://api.schedjoules.com/"
     let method: HTTPMethod = .get
     let parameters: Parameters = [:]
     let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
     
-    required init(resource: String) {
-        self.resource = resource
+    required init(path: String, queryItems: [URLQueryItem]) {
+        var urlComponents = URLComponents(string: host)
+        urlComponents?.path = path
+        urlComponents?.queryItems = queryItems
+        self.url = urlComponents?.url
```

Comment:
why do you use the `?` operator? Can `urlComponents` be `null` at this line? If so, why can that be? What's the expected behavior in that case?

Reply:
`URLComponents(string: host)` is a failable initalizer, thus why it returns an optional.

Reply:
That's too bad. Apparently Apple doesn't understand the concept of immutability.

Anyway, why do you continue when creating the URL failed? If the constructor failed you should raise an exception instead of moving on. It means that something went really wrong.